src/wlapplication.cc

```diff
@@ -211,6 +211,31 @@ bool is_autogenerated_and_expired(const std::string& filename,
 	return std::difftime(time(nullptr), mktime(&tfile)) > keep_time;
 }
 
+// A simple wrapper around the game logic thread. It ensures the thread is
+// always joined when this class goes out of scope. This is important when an
+// exception is thrown within the game anywhere, because without this, we will
+// not join the thread and this means the exception will never be caught and
+// printed, making debugging hard.
+// See
+// https://stackoverflow.com/questions/25397874/deleting-stdthread-pointer-raises-exception-libcabi-dylib-terminating
+class GameLogicThread {
+	public:
+		explicit GameLogicThread(bool* should_die) : should_die_(should_die), thread_(&UI::Panel::logic_thread) {}
+
+		GameLogicThread(const GameLogicThread&) = delete;
+		GameLogicThread& operator=(const GameLogicThread&) = delete;
```

Comment:
Better use the `DISALLOW_COPY_AND_ASSIGN` macro from base/macros.h for readability.